Assets/Scripts/CameraBob.cs

```diff
@@ -0,0 +1,55 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+// TODO: Add randomness
+// TODO: Add horizontal bob tilting
+// TODO: Increase bob intensity when sprinting
+// TODO: Make fps arms drift behind offset
+// TODO: Animate bob to a stop when player stops moving
+public class CameraBob : MonoBehaviour {
+    [SerializeField] private Camera cam;
+    [SerializeField] private float breathIntensity = 1;
+    [SerializeField] private float breathSpeed = 1;
+    [SerializeField] private float bobIntensity = 1;
+    [SerializeField] private float bobSpeed = 1;
+
+    private float camInitialLocalY;
+    private int breathFrameCounter = 0;
+    private int bobFrameCounter = 0;
+    public bool isBobing = false;
+    private bool isBreathing = true;
+    public float currentOffset = 0;
+
+
+    void OnValidate() {
+        breathIntensity = Mathf.Max(0, breathIntensity);
```

Comment:
You can use the `PositiveValueOnly` attribute from MyBox to do this too (I haven't installed the package yet though): https://github.com/Deadcows/MyBox/wiki/Attributes#maxvalue-minvalue-and-positivevalueonly

Reply:
Resolved